python/tk_desktop/desktop_engine_site_implementation.py

```diff
@@ -266,13 +265,18 @@ def run(self, splash, version, **kwargs):
         f.close()
         app.setStyleSheet(css)
 
-        # Initialize all of this after the style-sheet has been applied to any prompt are also
-        # styled after the Shotgun Desktop's visual-style.
-        splash.set_message("Initializing browser integration.")
-        try:
-            desktop_server_framework.launch_desktop_server(self._user.host, self._current_login["id"])
-        except Exception:
-            logger.exception("Unexpected error while trying to launch the browser integration:")
+        # Sadly, we can't tear down the previous server and restart it. Attempting to tear_down() and
+        # instantiate a new server will raise an error.ReactorNotRestartable exception. So we'll start
+        # our websocket integration only if there is no server running from the desktop startup.
+        if kwargs.get("server") is None:
```

Comment:
the comment doesn't mention why `server` is significant. If you could please elaborate that would be great!